linetools/isgm/emsystem.py

```diff
@@ -0,0 +1,536 @@
+        #self._trans = Table()
+        self._abund = Table()
+
+        # Refs (list of references)
+        self.Refs = []
+
+    def add_emlines_from_alis(self, alis_file, **kwargs):
+        '''
+        Code assumes the coordinates of the class (i.e. no checking)
+        Parameters
+        ----------
+        alis_file : str
+
+
+
+        Returns
+        -------
+
+        '''
+        dum_sys = EmSystem.from_alis(alis_file, self.coord)
+        # Add in lines
+        for emline in dum_sys._emlines:
+            self.add_emline(emline, **kwargs)
+
+    def add_emline(self, emline, tol=0.2*u.arcsec,
+                      chk_sep=True, chk_z=True, overlap_only=False,
+                      vtoler=1., debug=False, **kwargs):
+        """Add an EmLine object if it satisfies all of the rules.
+
+        For velocities, we demand that the new line has a velocity
+        range that is fully encompassed by the EmSystem.
+        We allow a small tolerance for round-off error
+
+        Should check for duplicates..
+
+        Parameters
+        ----------
+        emline : EmLine
+        tol : Angle, optional
+          Tolerance on matching coordinates
+          Only used if chk_sep=True
+        chk_sep : bool, optional
+          Perform coordinate check (expensive)
+        chk_z : bool, optional
+          Perform standard velocity range test
+        overlap_only : bool, optional
+          Only require that the components overlap in redshift
+        vtoler : float, optional
+          Tolerance for velocity in km/s
+
+        Returns
+        -------
+        test : bool
+          True if successful
+        """
+        # Coordinates
+        if chk_sep:
+            testcoord = bool(self.coord.separation(emline.attrib['coord']) < tol)
+        else:
+            testcoord = True
+        # Now redshift/velocity
+        if chk_z:
+            # Will avoid Quantity for speed
+            sys_vlim_mks = self.vlim.to('km/s').value
+            dz_toler = (1 + self.zem) * vtoler / c_mks
+            zlim_sys = self.zem + (1 + self.zem) * (sys_vlim_mks / c_mks)
+            testz = (emline.z >= (zlim_sys[0]-dz_toler)) & (
+                    emline.z <= (zlim_sys[1]+dz_toler))
+        else:
+            testz = True
+
+        # Additional checks (specific to EmSystem type)
+        testcomp = self.chk_emline(emline)
+        test = testcoord & testcomp & testz
+
+        # Append?
+        if test:
+            self._emlines.append(emline)
+        else:
+            warnings.warn('Input EmLine with wrest={} does not match EmSystem rules. Not appending'.format(emline.wrest))
+            if not testcoord:
+                warnings.warn('Failed coordinate match')
+            if not testcomp:
+                warnings.warn('Failed component check')
+            if not testz:
+                warnings.warn('Failed velocity overlap')
+        #
+        return test
+
+    def chk_emline(self, component):
```

Comment:
We don't use components with EmLine (at least not now).

Remove components and/or this method.  

Same goes for anywhere else you see components.